mmv1/third_party/terraform/resources/resource_compute_security_policy.go.erb

```diff
@@ -262,7 +262,7 @@ func resourceComputeSecurityPolicy() *schema.Resource {
 										Optional:     true,
 										Default:      "ALL",
 										Description:  `Determines the key to enforce the rateLimitThreshold on`,
-										ValidateFunc: validation.StringInSlice([]string{"ALL", "IP", "HTTP_HEADER", "XFF_IP", "HTTP_COOKIE"}, false),
+										ValidateFunc: validation.StringInSlice([]string{"ALL", "IP", "HTTP_HEADER", "XFF_IP", "HTTP_COOKIE", "HTTP_PATH", "SNI", "REGION_CODE", ""}, false),
```

Comment:
How volatile is this field? Would it make more sense to remove validation and allow the server to validate. We could link to external documentation within our docs to stay consistent with the valid fields.

Reply:
Hello there! The `ValidateFunc` was already develop before I set this new options. I think the `"HTTP_PATH", "SNI", "REGION_CODE"` are not the problem of this new change, if I remove this items my tests will fail and must be specified on prodivers, as you can see the example on screenshot without the items that you suggested to remove:

![image](https://user-images.githubusercontent.com/120479960/217053469-f20bed31-e560-4dae-a6e0-4496034439c5.png)

Reply:
Sorry I was suggesting removing the validation function altogether. Would allow the validation to happen at the api level instead of the terraform level!

Reply:
Hello there! I removed the `ValidateFunc` function on local environment and ran the test, it worked normally, but I kept the `enforce_on_key` value as an empty string, because the default attribute is still there.

![image](https://user-images.githubusercontent.com/120479960/217269565-8d28f75a-1555-4c9e-95fc-5a4774d43a52.png)

The screenshots bellow are the tests that I ran without `ValidateFunc`, but I kept the `enforce_on_key = ""` on the tests:

![image](https://user-images.githubusercontent.com/120479960/217277832-da81cf9b-b8d0-485d-ac32-d1bc992215e4.png)

![image](https://user-images.githubusercontent.com/120479960/217277895-4be436c4-b7b5-4428-bc49-3ef75b382890.png)

If you want that I remove the `ValidateFunc`, it'll get a breaking change on this PR. 

I would like to know if you want that I make this change or you guys will do this.

Reply:
> If you want that I remove the ValidateFunc, it'll get a breaking change on this PR.

you shouldn't get a breaking change notification! I think the change makes sense, but totally understand if you disagree !

Reply:
We undertand that the `ValidFunc` is important to validate the correct values setted on the attribute. Take this tests as an example,I removed the `ValidFunc` and I put "ANY" value on `enforce_on_key` to see if the API valited the correct values expected, the test failed, but the error is very strange than the expected, I would expect a message like "This enforce_on key_value is not correct" or something like this.

![image](https://user-images.githubusercontent.com/120479960/217354785-4a9040e4-bc84-4b07-88db-7eb36ca0e24d.png)

I'll put below the whole error:

```
==> Checking that code complies with gofmt requirements...
go vet
go generate  ./...
TF_ACC=1 TF_SCHEMA_PANIC_ON_ERROR=1 go test ./google-beta -v -run=TestAccComputeSecurityPolicy_withRateLimit_withEnforceOnKeyConfigs -timeout 240m -ldflags="-X=github.com/hashicorp/terraform-provider-google-beta/version.ProviderVersion=acc"
=== RUN   TestAccComputeSecurityPolicy_withRateLimit_withEnforceOnKeyConfigs
=== PAUSE TestAccComputeSecurityPolicy_withRateLimit_withEnforceOnKeyConfigs
=== CONT  TestAccComputeSecurityPolicy_withRateLimit_withEnforceOnKeyConfigs
    provider_test.go:315: Step 1/2 error: After applying this test step, the plan was not empty.
        stdout:
        
        
        Terraform used the selected providers to generate the following execution
        plan. Resource actions are indicated with the following symbols:
          ~ update in-place
        
        Terraform will perform the following actions:
        
          # google_compute_security_policy.policy will be updated in-place
          ~ resource "google_compute_security_policy" "policy" {
                id          = "projects/ciandt-cloudarmor-gcam/global/securityPolicies/tf-test-g0vspkc98l"
                name        = "tf-test-g0vspkc98l"
                # (5 unchanged attributes hidden)
        
              + rule {
                  + action      = "throttle"
                  + description = "default rule"
                  + preview     = (known after apply)
                  + priority    = 2147483647
        
                  + match {
                      + versioned_expr = "SRC_IPS_V1"
        
                      + config {
                          + src_ip_ranges = [
                              + "*",
                            ]
                        }
                    }
        
                  + rate_limit_options {
                      + conform_action = "allow"
                      + enforce_on_key = "ANY"
                      + exceed_action  = "redirect"
        
                      + exceed_redirect_options {
                          + target = "https://www.example.com"
                          + type   = "EXTERNAL_302"
                        }
        
                      + rate_limit_threshold {
                          + count        = 10
                          + interval_sec = 60
                        }
                    }
                }
              - rule {
                  - action      = "throttle" -> null
                  - description = "default rule" -> null
                  - preview     = false -> null
                  - priority    = 2147483647 -> null
        
                  - match {
                      - versioned_expr = "SRC_IPS_V1" -> null
        
                      - config {
                          - src_ip_ranges = [
                              - "*",
                            ] -> null
                        }
                    }
        
                  - rate_limit_options {
                      - ban_duration_sec = 0 -> null
                      - conform_action   = "allow" -> null
                      - exceed_action    = "redirect" -> null
        
                      - exceed_redirect_options {
                          - target = "https://www.example.com" -> null
                          - type   = "EXTERNAL_302" -> null
                        }
        
                      - rate_limit_threshold {
                          - count        = 10 -> null
                          - interval_sec = 60 -> null
                        }
                    }
                }
            }
        
        Plan: 0 to add, 1 to change, 0 to destroy.
--- FAIL: TestAccComputeSecurityPolicy_withRateLimit_withEnforceOnKeyConfigs (52.50s)
FAIL
FAIL    github.com/hashicorp/terraform-provider-google-beta/google-beta 52.518s
FAIL
make: *** [GNUmakefile:15: testacc] Erro 1
```

If I put back the `ValidFunc` and run the same test again, the result is this:

![image](https://user-images.githubusercontent.com/120479960/217353040-73a85d8d-5a98-4d23-a99b-5608d61880bc.png)

- Without the `ValidateFunc` it'll avoid any value inserted on `enforce_on_key` anyway, but it'll return a stranger message error than expected.

- With the `ValidateFunc` it'll still avoid any value inserted on `enforce_on_key`, but it'll return a menssage that make sense if the situation.

I think with the ValidateFunc will be helpful for showing the people where is the real problem.

That is my thought about this. It's up to you, if you think that is valid to remove I'll remove. The problem is the PR will get a breaking change and stop the process because I removed a solution that was already developed by other person and it was already merged. That is the only problem. 

Can you approve this PR with a breaking change warning? If yes, I'll remove the `ValidFunc` for you with no problem :smiley:

Reply:
Let's keep it then!! So sorry for the back and forth on it. :)

Thanks for your patience

Reply:
also could you pull the latest changes from main when you update the PR?

Reply:
Hello there! We pulled the latest changes, but we got some problems on:

- generate-diffs (graphite-docker-images) 
- run-rake-tests (graphite-docker-images) 

I noticed that other latests PRs got the same problem, we would like to know if it's an external problem.

Could you help us?

Reply:
Run /gcbrun .